unlock-app/app/frames/checkout/components/defaultFrame.tsx

```diff
@@ -1,11 +1,69 @@
 /* eslint-disable react/jsx-key */
 import React from 'react'
 import { Button } from 'frames.js/next'
-import { State } from '../frames'
+import { getAddressForFid } from 'frames.js'
 import { DefaultImage } from './DefaultImage'
+import { isMember as checkIsMember, getKeyPrice, checkAllowance } from './utils'
 
-export function getDefaultFrame(state: State) {
-  const lock = state.lock!
+export async function getDefaultFrame(ctx: any) {
+  const lock = ctx.state.lock!
+  const isErc20 = !!ctx.state.lock?.tokenAddress
+  const approved = ctx.searchParams.approved || lock.erc20Approved || !isErc20
+  let userAddress: string
+  const buttonText = approved
+    ? 'Mint'
+    : `Approve ${lock.currencySymbol} spending`
+
+  const fid = ctx.message?.requesterFid
+  if (fid && !lock.priceForUser) {
+    userAddress = await getAddressForFid({
+      fid,
+      options: { fallbackToCustodyAddress: true },
+    })
+
+    const { address: lockAddress, network, tokenAddress } = lock
+
+    const isMember = await checkIsMember(lockAddress, network, userAddress)
+    lock.isMember = isMember
+
+    const keyPrice = await getKeyPrice({
+      lockAddress,
+      network,
+      userAddress,
+    })
+    lock.priceForUser = keyPrice
+
+    if (tokenAddress) {
+      const allowance = await checkAllowance(
+        lockAddress,
+        Number(network),
+        userAddress as string,
+        tokenAddress!
+      )
+      if (Number(allowance) >= Number(keyPrice)) {
```

Comment:
That's not enough, the user may need to approve MORE than the keyprice to allow for renewals.